packages/hub/test/sessionflusher.test.ts

```diff
@@ -1,122 +1,128 @@
     const date = new Date('2021-04-08T12:18:23.043Z');
-    (flusher as any)._incrementSessionStatusCount('ok', date);
-    (flusher as any)._incrementSessionStatusCount('ok', date);
+    _incrementSessionStatusCount(flusher, 'ok', date);
+    _incrementSessionStatusCount(flusher, 'ok', date);
 
-    expect(sendSession).toHaveBeenCalledTimes(0);
+    expect(transporter).toHaveBeenCalledTimes(0);
 
     jest.advanceTimersByTime(61000);
 
-    expect(flusherFlushFunc).toHaveBeenCalledTimes(1);
-    expect(sendSession).toHaveBeenCalledWith(
+    expect(transporter).toHaveBeenCalledTimes(1);
+    expect(transporter).toHaveBeenCalledWith(
       expect.objectContaining({
         attrs: { release: '1.0.0', environment: 'dev' },
         aggregates: [{ started: '2021-04-08T12:18:00.000Z', exited: 2 }],
       }),
     );
   });
 
-  test('sendSessions is not called on flush if no sessions were captured', () => {
-    const flusher = new SessionFlusher(transport, { release: '1.0.0', environment: 'dev' });
-    const flusherFlushFunc = jest.spyOn(flusher, 'flush');
+  test('transporter is not called on flush if no sessions were captured', () => {
+    const transporter = makeTransporter();
+    new SessionFlusher(transporter, { release: '1.0.0', environment: 'dev' });
 
-    expect(sendSession).toHaveBeenCalledTimes(0);
+    expect(transporter).toHaveBeenCalledTimes(0);
     jest.advanceTimersByTime(61000);
-    expect(flusherFlushFunc).toHaveBeenCalledTimes(1);
-    expect(sendSession).toHaveBeenCalledTimes(0);
+    expect(transporter).toHaveBeenCalledTimes(1);
+    expect(transporter).toHaveBeenCalledTimes(0);
   });
 
   test('calling close on SessionFlusher should disable SessionFlusher', () => {
-    const flusher = new SessionFlusher(transport, { release: '1.0.x' });
-    flusher.close();
-    expect((flusher as any)._isEnabled).toEqual(false);
+    const transporter = makeTransporter();
+    const flusher = new SessionFlusher(transporter, { release: '1.0.x' });
+    closeSessionFlusher(flusher);
+    expect(flusher.isEnabled).toEqual(false);
   });
 
   test('calling close on SessionFlusher will force call flush', () => {
-    const flusher = new SessionFlusher(transport, { release: '1.0.x' });
-    const flusherFlushFunc = jest.spyOn(flusher, 'flush');
+    // GIVEN
+    const transporter = makeTransporter();
+    const flusher = new SessionFlusher(transporter, { release: '1.0.x' });
     const date = new Date('2021-04-08T12:18:23.043Z');
-    (flusher as any)._incrementSessionStatusCount('ok', date);
-    (flusher as any)._incrementSessionStatusCount('ok', date);
-    flusher.close();
 
-    expect(flusherFlushFunc).toHaveBeenCalledTimes(1);
-    expect(sendSession).toHaveBeenCalledWith(
+    // TODO: code-smell using internal function
+    // why can we call the public API instead of the internal one?
+    _incrementSessionStatusCount(flusher, 'ok', date);
```

Comment:
Ahhh interesting, the implementation of `incrementSessionStatusCount` goes back and forward and it is connected to global stuff ... probably such function shouldn't be in the session flusher but who calls the session flusher